### ISSUE TITLE
Static Site for iPads

### ISSUE THREAD

**Issue by halfempty** [stillmuseum/worksonpaper#110]
@swambold6 @dabernathy89

Sarah reports:

> We’re having connectivity issues on the iPads upstairs in the galleries. 
>
> I used Site Sucker to download the digital pub, successfully loaded the files on to one of the iPads and figured out how to point Kiosk Pro to it. (But) none of the images of the objects are loading. I’m guessing the reason is because Site Sucker missed them because they’re “locked up” in the modal windows.
>
> Seeing as this is a static site, do you have a complete folder of files that you could dropbox to me? I’m concerned this will not solve any problems, however, because the full website, including all the images is probably bigger than we can accommodate on the iPads, right? But maybe I can batch resize the images for optimum display on the iPads…?

I suspect the issue with the objects stems from the use of iFrames in that context.

Rather than Site Sucker, we can try using Simply Static (the static site generator) to create a fresh copy of the site, specifically for the iPads. The tricky part of this, to my mind, will be properly configuring the "destination." Ie. Can Simply Static comprehend the iPad filesystem?

That, along with the hard-drive issue you raise, may lead us to look for another option. I'm thinking of a web server inside the Museum's network. "Web server" may sound scary, but it's not really -- at the AGO I had an old PC under my desk that hosted part of the intranet, and we used it for things like this.

For option 1: **Dan**, will you test using the Simply Static plugin to generate a set of files that can be copied to an iPad and served via the Kiosk Pro app? (If you need to borrow an iPad for testing, I have a spare.)

For option 2: **Sarah**, is there already a computer that can act as an intranet webserver? If no, can we discuss what setting one up might entail?

**Comment by swambold6** [comment]
@halfempty @dabernathy89 The webserver had occurred to me too. Is it possible to partition an existing server for this use, or does it need to be a dedicated machine?

**Comment by halfempty** [comment]
@swambold6 **Sarah**, my first impression is that we should be able to configure the existing server. Should we skip "option 1" and go straight to "option 2"?

@dabernathy89 **Dan**, is configuring servers in this way something you have experience with?

**Comment by dabernathy89** [comment]
I don't have experience setting up any internal / intranet servers. However before we resort to that I can do some testing (we've got an iPad here at home) with the software you mentioned and the static site plugin.

**Comment by halfempty** [comment]
@dabernathy89 Thanks, Dan

**Comment by swambold6** [comment]
@halfempty @dabernathy89 All sounds good. I'll wait to hear from you, Dan, about your iPad tests. Thanks!

**Comment by halfempty** [comment]
@dabernathy89 Regarding the generation of static files, a few notes...

The first thing to do will be to got to Simply Static -> Settings and update the paths there. Then there are a couple of other steps to take:

- Deactivate the plugin "Members", which is used to restrict access to the dev site;
- Change the salts in wp-config to log-out anyone who happens to be working. The export hung once because of this;

Once we have the cocktail right and are ready to make the export for loading onto iPads, there's two more steps:

- Turn on Search Engine Visibility via Settings -> Reading; 
- Turn on Google Analytics by uncommenting lines in to theme files: `header.php` and `header-single.php`

And finally, after the export is complete, put these settings back.

**Comment by dabernathy89** [comment]
Just a quick update. Got the site downloaded and ported into my Dropbox - it's about 2 GB. I've been attempting to sync it with the Kiosk app but it's not clear whether it's working or not. I'm going to let it run overnight and hopefully know whether it worked or not in the morning.

**Comment by swambold6** [comment]
@dabernathy89 Can you send the dropbox link? Hoping to figure this out today, if possible...

**Comment by dabernathy89** [comment]
Sure thing. I am retrying the sync again now - couldn't get it working overnight. 2 GB may just be too much for the app to handle.

https://www.dropbox.com/sh/083h2k42vchflhn/AACfsCl81yt8JQ7Yc9foNlYga?dl=0

**Comment by swambold6** [comment]
@dabernathy89 Can we batch resize the images in Photoshop? I can't find where the object images are... ?

**Comment by swambold6** [comment]
@dabernathy89 So I found some of the object images and there are multiple sizes for different views. Can we toss the ones we don't need for iPad display? Will that decrease the file size?

**Comment by dabernathy89** [comment]
It looks like the bulk of the size is taken up by the largest image for each group, rather than the resized versions. I'm going to give batch compressing a shot and see how far that get us.

**Comment by swambold6** [comment]
@dabernathy89 I definitely have the space for a 2 GB file on the iPads. For some reason I can't get DB to download the file for me...

**Comment by dabernathy89** [comment]
I've just wrapped up the compression, however - I just tried running the site locally on my computer (while waiting for it to load on the iPad), and I'm still seeing the same issue with the modals. The static generator plugin has some options for change the way the URLs are structured in the export, I think that may be the trick.

**Comment by swambold6** [comment]
@dabernathy89 Sounds good. I was able to download everything and upload it to the iPads. Works beautifully, except the same problem as before...modal object images don't appear. :(

**Comment by dabernathy89** [comment]
Ok the modals in the new export are working correctly on my local install. Going to try to set it up on the iPad again.

**Comment by dabernathy89** [comment]
No luck using the new export on the iPad. Going to try one more method with the export.

**Comment by dabernathy89** [comment]
Alright I've figured out a temporary solution to bring it offline, but I had to do a find/replace on the exported files to get the iframes working. It should all be synced to the Dropbox folder now.

@halfempty - In the generated output for the iframes, it is referencing the image like this:

    src: './wp-content/uploads/[path to image]'

The filesystem is looking for the 'wp-content' folder inside of the folder where the iframe's HTML is generated. So the temporary fix is to do a find and replace to save all those iframe files like this:

    src: '../../wp-content/uploads/[path to image]'

Obviously going through this process every time there is an update isn't the best option. So we'll need to come up with something a little more automated.
